rskj-core/src/test/java/co/rsk/peg/BridgeSupportTest.java

```diff
@@ -323,6 +323,40 @@ public void getLockingCap() {
         verify(provider, times(1)).setLockingCap(constants.getInitialLockingCap());
     }
 
+    @Test
+    public void getActivePowpegRedeemScriptBeforeFork() {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(false);
+
+        Federation fed = bridgeConstants.getGenesisFederation();
+        FederationSupport federationSupport = mock(FederationSupport.class);
+        when(federationSupport.getActiveFederation()).thenReturn(fed);
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+                .withBridgeConstants(bridgeConstants)
+                .withActivations(activations)
+                .build();
+
+        assertEquals(null, bridgeSupport.getActivePowpegRedeemScript());
+    }
+
+    @Test
+    public void getActivePowpegRedeemScriptAfterFork() {
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP293)).thenReturn(true);
+
+        Federation fed = bridgeConstants.getGenesisFederation();
+        FederationSupport federationSupport = mock(FederationSupport.class);
```

Comment:
Take a look at the BridgeSupportBuilder, the object FederationSupport is built in the `build` method. If you want to mock the return value of `getActiveFederation()` then you should be mocking the BridgeStorageProvider. See the implementation of `getActiveFederation()` in `FederationSupport` class

Reply:
You are totally right.